### PR TITLE
IPython Notebook build target

### PR DESCRIPTION
**Pull request by haraldschilly**
I have the vague idea that it might be worthwhile to export Sphinx documents to the IPython Notebook format. It would be cool to be able to directly play around with the code in the documentation. I've only started experimenting around and the code in this PR  produces a few files for the sympy documentation (`make ipynb`). There is still a lot left to do and this is certainly **not** ready to be merged. Maybe someone is out there who wants to pick up this idea and flesh it out?

### CODE REVIEW

**Comment by birkenfeld** [comment]
Interesting idea for sure! If no one else picks it up, I will try to make some time to look at it.

**Comment by ashwin31** [comment]
@haraldschilly you should add a line near https://github.com/haraldschilly/sphinx/blob/ipynb/sphinx/quickstart.py#L687 list ipython

**Comment by shimizukawa** [comment]
@haraldschilly please fix test failures and please add unittest code for the builder.

**Comment by mgeier** [comment]
FYI, I made an extension that does the opposite: http://nbsphinx.readthedocs.org/.

**Comment by birkenfeld** [comment]
Won't be ready for 1.4.

File: sphinx/writers/ipynb.py [hunk 1]
```diff
@@ -0,0 +1,787 @@
...
+"""
+    sphinx.writers.text
+    ~~~~~~~~~~~~~~~~~~~
+
+    Custom docutils writer for IPython Notebooks.
+
+    :copyright: Copyright 2007-2015 by the Sphinx team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+import os
```
Comment (birkenfeld) [inline sphinx/writers/ipynb.py:11]: Codacy Issue found: [Unused import os](https://www.codacy.com/app/georg/sphinx/file/2138144138/issues/source?bid=3149936&fileBranchId=3149947#l11)

File: sphinx/writers/ipynb.py [hunk 2]
```diff
@@ -0,0 +1,787 @@
...
+        pass
+
+    def visit_table(self, node):
+        self.new_state("text")
+        self.table = [[]]
+
+    def depart_table(self, node):
+        lines = self.table[1:]
+        rows = []
+        colwidths = self.table[0]
```
Comment (birkenfeld) [inline sphinx/writers/ipynb.py:375]: Codacy Issue found: [Unused variable 'colwidths'](https://www.codacy.com/app/georg/sphinx/file/2138144138/issues/source?bid=3149936&fileBranchId=3149947#l375)

File: sphinx/writers/ipynb.py (hunk 1 above)
Comment (birkenfeld) [inline sphinx/writers/ipynb.py:11]: Codacy Issue found: [Unused import os](https://www.codacy.com/app/georg/sphinx/file/2138144138/issues/source?bid=3149936&fileBranchId=3149947#l11)

File: sphinx/writers/ipynb.py [hunk 3]
```diff
@@ -0,0 +1,787 @@
...
+    ~~~~~~~~~~~~~~~~~~~
+
+    Custom docutils writer for IPython Notebooks.
+
+    :copyright: Copyright 2007-2015 by the Sphinx team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+import os
+import re
+import textwrap
```
Comment (birkenfeld) [inline sphinx/writers/ipynb.py:13]: Codacy Issue found: [Unused import textwrap](https://www.codacy.com/app/georg/sphinx/file/2138144138/issues/source?bid=3149936&fileBranchId=3149947#l13)

File: sphinx/writers/ipynb.py (hunk 2 above)
Comment (birkenfeld) [inline sphinx/writers/ipynb.py:375]: Codacy Issue found: [Unused variable 'colwidths'](https://www.codacy.com/app/georg/sphinx/file/2138144138/issues/source?bid=3149936&fileBranchId=3149947#l375)

File: sphinx/writers/ipynb.py [hunk 4]
```diff
@@ -0,0 +1,787 @@
...
+    sphinx.writers.text
+    ~~~~~~~~~~~~~~~~~~~
+
+    Custom docutils writer for IPython Notebooks.
+
+    :copyright: Copyright 2007-2015 by the Sphinx team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+import os
+import re
```
Comment (birkenfeld) [inline sphinx/writers/ipynb.py:12]: Codacy Issue found: [Unused import re](https://www.codacy.com/app/georg/sphinx/file/2138144138/issues/source?bid=3149936&fileBranchId=3149947#l12)

File: sphinx/writers/ipynb.py (hunk 3 above)
Comment (birkenfeld) [inline sphinx/writers/ipynb.py:13]: Codacy Issue found: [Unused import textwrap](https://www.codacy.com/app/georg/sphinx/file/2138144138/issues/source?bid=3149936&fileBranchId=3149947#l13)

File: sphinx/writers/ipynb.py [hunk 5]
```diff
@@ -0,0 +1,787 @@
...
+import textwrap
+from itertools import groupby
+
+from nbformat import v4 as ipynb
+
+from six.moves import zip_longest
+
+from docutils import nodes, writers
+from docutils.utils import column_width
```
Comment (birkenfeld) [inline sphinx/writers/ipynb.py:21]: Codacy Issue found: [Unused column_width imported from docutils.utils](https://www.codacy.com/app/georg/sphinx/file/2138144138/issues/source?bid=3149936&fileBranchId=3149947#l21)

File: sphinx/writers/ipynb.py (hunk 4 above)
Comment (birkenfeld) [inline sphinx/writers/ipynb.py:12]: Codacy Issue found: [Unused import re](https://www.codacy.com/app/georg/sphinx/file/2138144138/issues/source?bid=3149936&fileBranchId=3149947#l12)

File: sphinx/writers/ipynb.py [hunk 6]
```diff
@@ -0,0 +1,787 @@
...
+
+    Custom docutils writer for IPython Notebooks.
+
+    :copyright: Copyright 2007-2015 by the Sphinx team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+import os
+import re
+import textwrap
+from itertools import groupby
```
Comment (birkenfeld) [inline sphinx/writers/ipynb.py:14]: Codacy Issue found: [Unused groupby imported from itertools](https://www.codacy.com/app/georg/sphinx/file/2138144138/issues/source?bid=3149936&fileBranchId=3149947#l14)

File: sphinx/writers/ipynb.py (hunk 5 above)
Comment (birkenfeld) [inline sphinx/writers/ipynb.py:21]: Codacy Issue found: [Unused column_width imported from docutils.utils](https://www.codacy.com/app/georg/sphinx/file/2138144138/issues/source?bid=3149936&fileBranchId=3149947#l21)

File: sphinx/writers/ipynb.py [hunk 7]
```diff
@@ -0,0 +1,787 @@
...
+                cells = []
+                for i, cell in enumerate(line):
+                    cells.append(cell)
+                rows.append(cells)
+
+        def writerow(row):
+            lines = zip_longest(*row)
+            for line in lines:
+                out += "<tr>"
+                for i, cell in enumerate(line):
```
Comment (birkenfeld) [inline sphinx/writers/ipynb.py:390]: Codacy Issue found: [Unused variable 'i'](https://www.codacy.com/app/georg/sphinx/file/2138144138/issues/source?bid=3149936&fileBranchId=3149947#l390)

File: sphinx/writers/ipynb.py (hunk 6 above)
Comment (birkenfeld) [inline sphinx/writers/ipynb.py:14]: Codacy Issue found: [Unused groupby imported from itertools](https://www.codacy.com/app/georg/sphinx/file/2138144138/issues/source?bid=3149936&fileBranchId=3149947#l14)

File: sphinx/writers/ipynb.py (hunk 7 above)
Comment (birkenfeld) [inline sphinx/writers/ipynb.py:390]: Codacy Issue found: [Unused variable 'i'](https://www.codacy.com/app/georg/sphinx/file/2138144138/issues/source?bid=3149936&fileBranchId=3149947#l390)

**Comment by tk0miya** [comment]
I agree ipython notebook is great software.
But, at present, sphinx project doesn't have enough time to maintain a new builder.
Now we're going to keep sphinx-core simple and slim.

In addition, our next major release; Sphinx-1.6 supports to load the 3rd party builders using entry points (refs: #2803).

So I feel it is not needed to merge the builder into sphinx-core.

**Comment by shimizukawa** [comment]
+1 to close this PR.

**Pull request closed without merging**